app/models/mixins/supports_feature_mixin.rb

```diff
@@ -99,24 +87,46 @@ def unsupported
   class_methods do
     # This is the DSL used a class level to define what is supported
     def supports(feature, &block)
-      define_supports_feature_methods(feature, &block)
+      self.supports_features = supports_features.merge(feature.to_sym => block || true)
     end
 
     # supports_not does not take a block, because its never supported
     # and not conditionally supported
     def supports_not(feature, reason: nil)
-      define_supports_feature_methods(feature, :is_supported => false, :reason => reason)
+      self.supports_features = supports_features.merge(feature.to_sym => reason.presence || false)
     end
 
     # query the class if the feature is supported or not
     def supports?(feature)
-      method_name = "supports_#{feature}?"
-      if respond_to?(method_name)
-        public_send(method_name)
-      else
-        unsupported_reason_add(feature)
-        false
+      check_supports(feature.to_sym, :instance => self)
+    end
+
+    def check_supports(feature, instance:)
```

Comment:
super-minor but is this intended to be a public or private interface?  If private, it should probably be marked as such.

Reply:
it is private. thanks

Reply:
Yes, in the next pr we drop this method.